### ISSUE TITLE
setting up the  Configuration file  for a custom Data set

### ISSUE THREAD

**Issue by kadimaolivier** [facebookresearch/PyTorch-BigGraph#100]
### **## Steps to **reproduce****

(base) kadimas-MacBook-Pro:evalu kadima$ torchbiggraph_import_from_tsv \
>   --lhs-col=0 --rel-col=1 --rhs-col=2 \
>  config.py \
>  data/LAKE/DB_Lake.db2013.txt 

Looking up relation types in the edge files...
- Found 6 relation types
- Removing the ones with fewer than 1 occurrences...
- Left with 6 relation types
- Shuffling them...
Searching for the entities in the edge files...
Entity type all:
- Found 31 entities
- Removing the ones with fewer than 1 occurrences...
- Left with 31 entities
- Shuffling them...
Preparing entity path data/LAKE:
- Writing count of entity type all and partition 0
- Writing count of dynamic relations
Preparing edge path data/LAKE/DB_Lake.db2013_partitioned, out of the edges found in data/LAKE/DB_Lake.db2013.txt
- Edges will be partitioned in 1 x 1 buckets.
- Processed 43 edges in total
- Writing bucket (0, 0), containing 43 edges...

(base) kadimas-MacBook-Pro:evalu kadima$ torchbiggraph_train \
> config.py \
> -p edge_paths=data/LAKE/DB_Lake.db2013_partitioned

2019-09-11 15:42:43  Loading entity counts...
2019-09-11 15:42:43  Creating workers...
2019-09-11 15:42:43  Initializing global model...
2019-09-11 15:42:43  Starting epoch 1 / 3 edge path 1 / 1 edge chunk 1 / 1
2019-09-11 15:42:43  edge_path= data/LAKE/DB_Lake.db2013_partitioned
still in queue: 0
2019-09-11 15:42:43  Swapping partitioned embeddings None ( 0 , 0 )
2019-09-11 15:42:43  Loading entities
2019-09-11 15:42:45  ( 0 , 0 ): bucket 1 / 1 : Processed 43 edges in 2.11 s ( 2e-05 M/sec ); io: 0.02 s ( 0.05 MB/sec )
2019-09-11 15:42:45  ( 0 , 0 ): loss:  15.694 , violators_lhs:  487.14 , violators_rhs:  507.163 , count:  43
2019-09-11 15:42:45  Swapping partitioned embeddings ( 0 , 0 ) None
2019-09-11 15:42:45  Writing partitioned embeddings
2019-09-11 15:42:45  Finished epoch 1 path 1 pass 1; checkpointing global state.
2019-09-11 15:42:45  My rank: 0
2019-09-11 15:42:45  Writing metadata...
2019-09-11 15:42:45  Writing the checkpoint...
2019-09-11 15:42:45  Switching to new checkpoint version...
2019-09-11 15:42:45  Starting epoch 2 / 3 edge path 1 / 1 edge chunk 1 / 1
2019-09-11 15:42:45  edge_path= data/LAKE/DB_Lake.db2013_partitioned
still in queue: 0
2019-09-11 15:42:45  Swapping partitioned embeddings None ( 0 , 0 )
2019-09-11 15:42:45  Loading entities
2019-09-11 15:42:45  ( 0 , 0 ): bucket 1 / 1 : Processed 43 edges in 0.07 s ( 0.00066 M/sec ); io: 0.00 s ( 0.36 MB/sec )
2019-09-11 15:42:45  ( 0 , 0 ): loss:  30.0024 , violators_lhs:  435.395 , violators_rhs:  473.442 , count:  43
2019-09-11 15:42:45  Swapping partitioned embeddings ( 0 , 0 ) None
2019-09-11 15:42:45  Writing partitioned embeddings
2019-09-11 15:42:45  Finished epoch 2 path 1 pass 1; checkpointing global state.
2019-09-11 15:42:45  My rank: 0
2019-09-11 15:42:45  Writing metadata...
2019-09-11 15:42:45  Writing the checkpoint...
2019-09-11 15:42:45  Switching to new checkpoint version...
2019-09-11 15:42:45  Starting epoch 3 / 3 edge path 1 / 1 edge chunk 1 / 1
2019-09-11 15:42:45  edge_path= data/LAKE/DB_Lake.db2013_partitioned
still in queue: 0
2019-09-11 15:42:45  Swapping partitioned embeddings None ( 0 , 0 )
2019-09-11 15:42:45  Loading entities
2019-09-11 15:42:45  ( 0 , 0 ): bucket 1 / 1 : Processed 43 edges in 0.06 s ( 0.00069 M/sec ); io: 0.00 s ( 0.37 MB/sec )
2019-09-11 15:42:45  ( 0 , 0 ): loss:  17.278 , violators_lhs:  370.86 , violators_rhs:  353.07 , count:  43
2019-09-11 15:42:45  Swapping partitioned embeddings ( 0 , 0 ) None
2019-09-11 15:42:45  Writing partitioned embeddings
2019-09-11 15:42:45  Finished epoch 3 path 1 pass 1; checkpointing global state.
2019-09-11 15:42:45  My rank: 0
2019-09-11 15:42:45  Writing metadata...
2019-09-11 15:42:46  Writing the checkpoint...
2019-09-11 15:42:46  Switching to new checkpoint version...
2019-09-11 15:42:46  Exiting
(base) kadimas-MacBook-Pro:evalu kadima$ 

(base) kadimas-MacBook-Pro:evalu kadima$ torchbiggraph_export_to_tsv   --dict data/LAKE/dictionary.json   --checkpoint model/lake  --out entity_embeddings.tsv    --out relation_types_parameters.tsv

Loading relation types and entities...
Loading model check point...
Writing embeddings for entity type all partition 0...
- Processed 31 entities in total
Writing relation parameters...
Done exporting data to relation_types_parameters.tsv
 

**## Observed Results**
- I have followed all the step provided in order to get embeddings which i have managed to get embeddings but i have observed that the result are  changing  when i changed my dimension size in the configuration file 

**Questions**

- is there a way or rule to define **the dimension** in my configuration file (# score model) ?

- is there any strategy way to set up **he dimension** in my configuration file  according to my data set? if yes,may you please explain the rules or strategy to set up the dimension in my config file

  * What happened?  This could be a description, log output, etc.

## Expected Results

  * What did you expect to happen?
 i was expected after getting my embeddings ,being able to calculate the euclidean distance between my embeddings and find out the simularity between entities  but i realised when i change my dimension in my configuration file it does affect the results and semanticaly it does not provid any matching or proximity or simularity between entities...

## Relevant Code

  ```
  // TODO(you): code here to reproduce the problem
  ```

**Comment by lerks** [comment]
The dimension of the embeddings that are trained is specified using the `dimension` parameter in the config. There is no default so you always have to explicitly specify it. A different value for the dimension will lead to a different quality of the model, as a low dimension may underfit whereas a high one may overfit. Finding the right dimension is part of the hyperparameter tuning that you need to do for your model. There's no one-size-fits-all.

**Comment by lerks** [comment]
Let me also say that the graph you're training on is extremely small and it will be hard for you to learn meaningful embeddings from it. You have 31 entities and 46 edges which gives you an average degree of 3. Ideally we recommend entities to have tens of edges each. Depending on what you're trying to achieve, there may be other tools more suited to the scale you're working at.

**Comment by kadimaolivier** [comment]
Thank you very much for all the information provided....i really appreciate.

sincerely yours.

Olivier
